### PR TITLE
kv,roachpb,storage: various log/trace message improvements

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

**Comment by petermattis** [comment]
Review status: 0 of 10 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/storage/replica_proposal.go, line 390 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/15556#-Kj4GdWMiHmX2PlqG4h9:-Kj4GdWMiHmX2PlqG4hA:b1sa8vf) ([raw file](https://github.com/cockroachdb/cockroach/blob/5438a7c53370079797627c5f9849d954a7eea22d/pkg/storage/replica_proposal.go#L390)):*
> ```Go
> 		// infrequent.
> 		if newLease.Type() == roachpb.LeaseEpoch || log.V(1) && leaseChangingHands {
> 			log.Infof(ctx, "new range lease %s following %s", newLease, prevLease)
> ```

Will this be too verbose? My expectation is that epoch-based leases change very infrequently. Though during a chaos event there will be a lot of messages.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15556)*

**Comment by tamird** [comment]
:lgtm_strong: 

---

Reviewed 1 of 1 files at r1, 5 of 5 files at r2, 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[pkg/kv/transport.go, line 287 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/15556#-Kj4HW6Dd82ETb6t-Idb:-Kj4HW6Dd82ETb6t-Idc:b-470ox2) ([raw file](https://github.com/cockroachdb/cockroach/blob/5438a7c53370079797627c5f9849d954a7eea22d/pkg/kv/transport.go#L287)):*
> ```Go
> 
> func (gt *grpcTransport) NextReplica() roachpb.ReplicaDescriptor {
> 	if gt.clientIndex >= len(gt.orderedClients) {
> ```

could you add a comment explaining why this doesn't call `IsExhausted`? Looks like it's something to do with `maybeResurrectRetryables`, but I don't have enough of this paged in to infer if this is correct or if that call is needed here as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15556#-:-Kj4HOvSDTcHp_4be0va:b-uwkowd)*

File: pkg/storage/replica_proposal.go [hunk 1]
```diff
@@ -383,6 +383,14 @@ func (r *Replica) leasePostApply(
 	iAmTheLeaseHolder := newLease.Replica.ReplicaID == replicaID
 	leaseChangingHands := prevLease.Replica.StoreID != newLease.Replica.StoreID
 
+	if iAmTheLeaseHolder {
+		// Always log lease acquisition for epoch-based leases which are
+		// infrequent.
+		if newLease.Type() == roachpb.LeaseEpoch || log.V(1) && leaseChangingHands {
```
Comment (spencerkimball) [inline pkg/storage/replica_proposal.go]: Could you employ some parentheses to clarify the intended operator precedence?

**Review by spencerkimball — APPROVED** [review]

**Comment by petermattis** [comment]
Review status: 0 of 10 files reviewed at latest revision, 3 unresolved discussions.

---

*[pkg/kv/transport.go, line 287 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/15556#-Kj4HW6Dd82ETb6t-Idb:-Kj4UMr7FAFGtlcndWUT:b47f84l) ([raw file](https://github.com/cockroachdb/cockroach/blob/5438a7c53370079797627c5f9849d954a7eea22d/pkg/kv/transport.go#L287)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

could you add a comment explaining why this doesn't call `IsExhausted`? Looks like it's something to do with `maybeResurrectRetryables`, but I don't have enough of this paged in to infer if this is correct or if that call is needed here as well.
</blockquote></details>

Calling `IsExhausted` is more correct. Done.

---

*[pkg/storage/replica_proposal.go, line 389 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/15556#-Kj4GT2t0R7YBmtDyqdc-r3-389:-Kj4URzLfv2KJwMz5Oo8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5438a7c53370079797627c5f9849d954a7eea22d/pkg/storage/replica_proposal.go#L389)):*
<details><summary><i>Previously, spencerkimball (Spencer Kimball) wrote…</i></summary><blockquote>

Could you employ some parentheses to clarify the intended operator precedence?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15556)*

**Comment by tamird** [comment]
Reviewed 1 of 10 files at r4, 4 of 5 files at r5, 5 of 5 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15556)*

**Pull request merged**